File: scripts/ui/ANIui.js
const crashes = {length: 0, latest: null},
      ANIlinks = require("ui/ANIlinks"),
      RedirectURLDialog = require("ui/dialogs/RedirectURLDialog");
      
module.exports = {
    reportDialog: new RedirectURLDialog("report", () => {
        return "https://github.com/FlinTyX/Animalia/issues/new?assignees=&labels=bug&body=" + Strings.encode(Strings.format(
            "# Crash Report:" + "\n" +
            "**Describe Your Issue**: ..." + "\n\n" +
            "**Platform**: `@`" + "\n" +
            "**Version**: `@`" + ", `Animalia @`" + "\n" +
            "**Latest Crash**: `@`" + "\n\n" +
            "Thanks for reporting your issue! It will be fixed as soon as possible.",

            OS.isAndroid ? "Android " + Core.app.getVersion() : (OS.osName + " x" + OS.osArchBits),
            Version.combined(), 
            Vars.mods.getMod("animalia").meta.version,
            crashes.latest, //how can i read whats inside it?
        ));
            
    }),

    cataclysmFrag: require("ui/fragments/CataclysmFragment"),

    addStats(stats, main, keepStat, values){
        stats.add(main, extend(StatValue, {
            display(table){
                table.clear();
                table.left();
                
                table.table(null, t => {
                    values.forEach(e => {
                        t.add(e).left();
                        t.row();
                    });

                    if(keepStat){
                        t.row();
                        t.table(null, tt => {
                            let stat = stats.toMap().get(main.category).get(main);
                            tt.add("[lightgray]" + main.localized() + (main.localized().endsWith(":") ? " " : ":[] "));

                            if(stat.size < 2){
                                tt.add(values[0]).left();
                            } else {
                                stat.get(0).display(tt);
                                tt.left();
                            }

                            tt.pack();
                        }).growX().left();
                    }
                }).fillX();
            }
        }));
    },

    statValue(stat, value){
        return "[lightgray]" + Core.bundle.get(stat.toString()) + (stat.toString().endsWith(":") ? " " : ":[] ") + value.toString();
    },

    statValueBundle(stat, value){
        return module.exports.statValue(stat, Core.bundle.get(value));
    },

    init(){
        module.exports.setupRules();
        module.exports.cataclysmFrag.build(Vars.ui.hudGroup);

        ANIlinks.setup();

        Vars.ui.settings.graphics.checkPref(
            "crashreport", 
            Core.settings.getBool("crashreport"),
            bool => Core.settings.put("crashreport", bool)
        );
    },

    setupRules(){
        crashes.length = Core.settings.getDataDirectory().child("crashes").file().listFiles().length;
        crashes.latest = Core.settings.getDataDirectory().child("crashes").file().listFiles()[0];

        if(!Core.settings.has("crashcounter")){
            Core.settings.put("crashcounter", new Packages.java.lang.Integer(crashes.length));
        }

        if(!Core.settings.has("crashreport")){
            Core.settings.put("crashreport", true);
        }

        if(Core.settings.getInt("crashcounter") > 0 && Core.settings.getInt("crashcounter") < crashes.length && Core.settings.getBool("crashreport")){
            module.exports.reportDialog.build();

            Core.settings.put("crashcounter", new Packages.java.lang.Integer(crashes.length));
        }
    }
};

Events.on(ClientLoadEvent, () => {
    module.exports.init();
});

File: src/animalia/ui/AniUI.java
package animalia.ui;

import animalia.ui.dialogs.ReportDialog;
import arc.Core;
import arc.struct.Seq;
import mindustry.Vars;
import animalia.ui.fragments.CataclysmFragment;
import animalia.ui.fragments.TemperatureFragment;
import mindustry.world.meta.Stat;
import mindustry.world.meta.StatValue;
import mindustry.world.meta.Stats;

public class AniUI {
    public static CataclysmFragment cataclysmFrag;
    public static TemperatureFragment tempFrag;

    public static ReportDialog reportDialog;

    public AniUI(){

    }

    public static void load(){
        cataclysmFrag = new CataclysmFragment();
        tempFrag = new TemperatureFragment();
        reportDialog = new ReportDialog();

        cataclysmFrag.build(Vars.ui.hudGroup);
        tempFrag.build(Vars.ui.hudGroup);

        reportDialog.load();

        AniLinks.load();

        if(!Core.settings.has("reportcrash")) Core.settings.put("reportcrash", true);
        if(!Core.settings.has("entryweathers")) Core.settings.put("entryweathers", true);

        Vars.ui.settings.game.checkPref(
            "reportcrash",
            Core.settings.getBool("reportcrash"),
            bool -> Core.settings.put("reportcrash", bool)
        );

        Vars.ui.settings.game.checkPref(
            "entryweathers",
            Core.settings.getBool("entryweathers"),
            bool -> Core.settings.put("entryweathers", bool)
        );
    }

    public static <T> void addStats(Stats stats, Stat main, boolean keepStat, String... values){
        stats.add(main, table -> {
            table.clear();
            table.left();

            table.table(t -> {
                for(String e : values) {
                    t.add(e).left();
                    t.row();
                }

                if (keepStat) {
                    t.row();
                    t.table(tt -> {
                        Seq<StatValue> stat = stats.toMap().get(main.category).get(main);

                        tt.add("[lightgray]" + main.localized() + (main.localized().endsWith(":") ? " " : ":[] "));

                        if (stat.size < 2) {
                            t.add(values[0]).left();
                        } else {
                            stat.get(0).display(tt);
                            tt.left();
                        }

                        tt.pack();
                    }).growX().left();
                }
            }).fillX();
        });
    }

    public static String statValue(String stat, String value){
        return "[lightgray]" + Core.bundle.get(stat) + (stat.endsWith(":") ? " " : ":[] ") + value;
    }

    public static String statValueBundle(String stat, String value){
        return statValue(stat, Core.bundle.get(value));
    }
}
